test/functional/editor/jump_spec.lua

```diff
@@ -137,3 +139,88 @@ describe("jumpoptions=stack behaves like 'tagstack'", function()
        exec_capture('jumps'))
   end)
 end)
+
+describe("jumpoptions=view", function()
+  local file1 = 'Xtestfile-functional-editor-jumps'
+  local file2 = 'Xtestfile-functional-editor-jumps-2'
+  local function content()
+    local c = {}
+    for i=1,30 do
+      c[i] = i .. " line"
+    end
+    return table.concat(c, "\n")
+  end
+  before_each(function()
+    clear()
+    write_file(file1, content(), false, false)
+    write_file(file2, content(), false, false)
+    command('set jumpoptions=view')
+  end)
+  after_each(function()
+    os.remove(file1)
+    os.remove(file2)
+  end)
+
+  it('restores the view', function()
+    local screen = Screen.new(5, 8)
+    screen:attach()
+    command("edit " .. file1)
+    feed("12Gztj")
+    feed("gg<C-o>")
+    screen:expect([[
+    12 line     |
+    ^13 line     |
+    14 line     |
+    15 line     |
+    16 line     |
+    17 line     |
+    18 line     |
+                |
+    ]])
+  end)
+
+  it('restores  the view across files', function()
+    local screen = Screen.new(5, 5)
+    screen:attach()
+    command("args " .. file1 .. " " .. file2)
+    feed("12Gzt")
+    command("next")
+    feed("G")
+    screen:expect([[
+    27 line     |
+    28 line     |
+    29 line     |
+    ^30 line     |
+                |
+    ]])
+    feed("<C-o><C-o>")
+    screen:expect([[
+    ^12 line     |
+    13 line     |
+    14 line     |
+    15 line     |
+                |
+    ]])
+  end)
+
+  it('fallback to standard behavior when view can\'t be recovered', function()
+    local screen = Screen.new(5, 8)
+    screen:attach()
+    command("edit " .. file1)
+    feed("7GzbG")
+    curbufmeths.set_lines(0, 2, true, {})
+    -- Since lines where deleted can't se a 7 lines topline offset, so it
+    -- fallback to centering the view.
```

Comment:
Can you clarify this comment? I don't follow, specifically "can't se a 7 lines topline offset"

Reply:
I think this explanation it's a little better https://github.com/muniter/neovim/blob/4897a954bc06b3d6935b53b783285f5994e48f45/test/functional/editor/jump_spec.lua#L212